File: data/derived_cot/rq1_traces/translation/r1/java_to_cpp/code_output/SplitSentence.cpp
#include <vector>
#include <string>
#include <sstream>
#include <cctype>

class SplitSentence {
public:
    std::vector<std::string> splitSentences(const std::string& s);
    int countWords(const std::string& sentence);
    int processTextFile(const std::string& sentencesString);

private:
    bool isWordChar(char c) {
        return (c >= 'a' && c <= 'z') || (c >= 'A' && c <= 'Z') || (c >= '0' && c <= '9') || (c == '_');
    }
    bool isUpperCase(char c) {
        return (c >= 'A' && c <= 'Z');
    }
    bool isLowerCase(char c) {
        return (c >= 'a' && c <= 'z');
    }
};

std::vector<std::string> SplitSentence::splitSentences(const std::string& s) {
    std::vector<std::string> sentences;
    if (s.empty()) {
        return sentences;
    }
    int lastEnd = 0;
    int len = s.length();

    for (int i = 0; i < len - 1; i++) {
        if (s[i] == '.' || s[i] == '?') {
            if (std::isspace(static_cast<unsigned char>(s[i+1]))) {
                bool skip = false;
                if (s[i] == '.') {
                    if (i - 3 >= 0) {
                        if (isWordChar(s[i-3]) && s[i-2] == '.' && isWordChar(s[i-1]) && s[i] == '.') {
                            skip = true;
                        }
                    }
                    if (!skip && i - 2 >= 0) {
                        if (isUpperCase(s[i-2]) && isLowerCase(s[i-1]) && s[i] == '.') {
                            skip = true;
                        }
                    }
                }
                if (!skip) {
                    sentences.push_back(s.substr(lastEnd, i - lastEnd + 1));
                    lastEnd = i + 2;
                }
            }
        }
    }

    if (lastEnd < len) {
        sentences.push_back(s.substr(lastEnd));
    }

    return sentences;
}

int SplitSentence::countWords(const std::string& sentence) {
    std::string cleaned;
    for (char c : sentence) {
        if ((c >= 'a' && c <= 'z') || (c >= 'A' && c <= 'Z') || std::isspace(static_cast<unsigned char>(c))) {
            cleaned += c;
        }
    }

    std::istringstream iss(cleaned);
    std::string word;
    int count = 0;
    while (iss >> word) {
        count++;
    }
    return count;
}

int SplitSentence::processTextFile(const std::string& sentencesString) {
    auto sentences = splitSentences(sentencesString);
    int maxCount = 0;
    for (const auto& sentence : sentences) {
        int cnt = countWords(sentence);
        if (cnt > maxCount) {
            maxCount = cnt;
        }
    }
    return maxCount;
}

File: data/derived_cot/rq2_eval/translation/r1/_converted_code/java_to_py/SplitSentence.py
import re

class SplitSentence:

    def __init__(self):
        self.sentence_pattern = re.compile('(?<!\\w\\.\\w.)(?<![A-Z][a-z]\\.)(?<=\\.|\\?)\\s')

    def split_sentences(self, sentencesString):
        sentences = []
        last_end = 0
        for match in self.sentence_pattern.finditer(sentencesString):
            start_index = last_end
            end_index = match.end() - 1
            sentences.append(sentencesString[start_index:end_index])
            last_end = match.end()
        if last_end < len(sentencesString):
            sentences.append(sentencesString[last_end:])
        return sentences

    def count_words(self, sentence):
        cleaned = re.sub('[^a-zA-Z\\s]', '', sentence)
        if cleaned == '':
            return 1
        words = re.split('\\s+', cleaned)
        while words and words[-1] == '':
            words.pop()
        return len(words)

    def process_text_file(self, sentencesString):
        sentences = self.split_sentences(sentencesString)
        max_count = 0
        for sentence in sentences:
            count = self.count_words(sentence)
            if count > max_count:
                max_count = count
        return max_count